### PR TITLE
use Slika for image resizing and cropping

### PR DESCRIPTION
**Pull request by splitbrain**
This replaces our own resize/crop functions by the Slika library.

This is not yet ideal, as there is quite a bit duplicated code between media_resize and media_crop now. Ideally these two should be replaced by a single method handling both. I'm just not sure where to best put it yet.

Using Slika introduces two new features:

* auto rotation based on EXIF #3059
* support for webp #3238

### CODE REVIEW

File: inc/media.php [hunk 1]
```diff
@@ -2055,40 +2055,37 @@ function media_nstree_li($item){
...
-        media_resize_imageGD($ext, $file, $info[0], $info[1], $local, $w, $h)
-    ) {
-        if($conf['fperm']) @chmod($local, $conf['fperm']);
-        return $local;
+    $options = [
+        'quality' => $conf['jpg_quality'],
+        'imconvert' => $conf['im_convert'],
+    ];
+
+    if( $mtime > @filemtime($file) ) {
```
Comment (Frogging101) [inline inc/media.php:2071]: This is broken. It should be `$mtime <= (int) @filemtime($file)`. See my patch here: https://github.com/splitbrain/dokuwiki/pull/3259

**Review by Frogging101 — CHANGES_REQUESTED** [review]

File: inc/media.php (hunk 1 above)
Comment (splitbrain) [reply inc/media.php]: hah. you're right of course. Thank you

**Comment by Frogging101** [comment]
Thanks for doing this also, @splitbrain. Just yesterday I was creating an image gallery on my wiki and it didn't look too good because of the upside-down images. Then I found this PR and it fixed it :+1:

**Comment by bannisdale** [comment]
This should also realize https://github.com/splitbrain/dokuwiki/issues/2476, right? :)

**Comment by splitbrain** [comment]
@bannisdale good question. Slika supports percentages when resizing. We might need to adjust a few input checks in the rest of DokuWiki to expose that though. Haven't tried ;-)

**Comment by bannisdale** [comment]
Imho, it would definitely be good to check/realize this when the corresponding objects are currently being modified anyway. :D

**Comment by splitbrain** [comment]
Adding percentage support isn't trivial because we also use the value to create a width attribute on the image. But we can't set the width to "50%" when we scale down the image as well as the result would be 25% then. I will merge this and percentages can be added in another PR.

**Pull request merged**